libpod/networking_slirp4netns.go

```diff
@@ -314,9 +314,53 @@ func (r *Runtime) setupSlirp4netns(ctr *Container) error {
 		}
 		return r.setupRootlessPortMappingViaRLK(ctr, netnsPath)
 	}
+
+	// Calculate Slirp addresses now that we are pretty sure the command executed
+	if netOptions.cidr != "" {
+		ipv4, ipv4network, err := net.ParseCIDR(netOptions.cidr)
+		if err != nil || ipv4.To4() == nil {
+			return errors.Errorf("invalid cidr %q", netOptions.cidr)
+		}
+		expectedIP, err := addToIP(ipv4, ipv4network, uint32(100))
+		if err != nil {
+			return errors.Wrapf(err, "error calculating expected ip for slirp4netns")
+		}
+		expectedGatewayIP, err := addToIP(ipv4, ipv4network, uint32(2))
+		if err != nil {
+			return errors.Wrapf(err, "error calculating expected gateway ip for slirp4netns")
+		}
+		expectedDNSIP, err := addToIP(ipv4, ipv4network, uint32(3))
+		if err != nil {
+			return errors.Wrapf(err, "error calculating expected dns ip for slirp4netns")
+		}
+		// Save off calculated address into globals. I don't like this either
+		slirp4netnsIP = expectedIP.String()
```

Comment:
I don't think we should use globals, this will fail when used with the podman service.

Reply:
Okay, I'll look at moving away from the globals. I think when I initially wrote this those were constant values that I changed to variables so I didn't even think about if they **should** be globals. Thanks for catching that :)

Reply:
I think you can add a new field (maybe called slirp4netnsSubnet) to the container struct here https://github.com/containers/podman/blob/663ea964733f5d7ccef9b457b33e97ebdec79e70/libpod/container.go#L97
and add methods like GetSlirp4netnsIP, GetSlirp4netnsGateway and GetSlirp4netnsDNS which returns the correct ip calculated from the stored subent.

Reply:
Fixed: https://github.com/containers/podman/commit/3ce696e05426bc03a541ba6f36f8ee48080c0c66#diff-b50e7e875575fcd700a1431a314b378fc179af9a4d1da93cbb62a42d10949c0aR335

I made the helper functions generatl libpod functions that take subnets instead of Container functions that reference internal struct members because I didn't think it was a container specific thing. I also essentially reverted your changes here: https://github.com/containers/podman/commit/f99b7a314bf6d285b92a528d1a0800c35dfdc603 since I had a merge conflict and my changes covered what you did with that commit as far as I could tell.